src/model/liveselection.js

```diff
@@ -0,0 +1,674 @@
+/**
```

Comment:
I would make `LiveSelection` private or protected. It should be used only by the `DocumentSelection` and could be even defined there.

Reply:
Both classed could go to the same file.